modules/nwtc-library/src/NWTC_C_Binding.f90

```diff
@@ -49,7 +49,73 @@ SUBROUTINE SetErr(ErrStat, ErrMsg, ErrStat_C, ErrMsg_C)
     else
         ErrMsg_C = TRANSFER( TRIM(ErrMsg)//C_NULL_CHAR, ErrMsg_C )
     endif
-END SUBROUTINE SetErr
+END SUBROUTINE SetErrStat_F2C
+
+!> This subroutine incorporates the local error status and error messages into the global error
+!! status and message. It expects both local and global error messages to be null-terminated
+!! C strings.
+SUBROUTINE SetErrStat_C(ErrStatLocal, ErrMessLocal, ErrStatGlobal, ErrMessGlobal, RoutineName)
+
+    INTEGER(C_INT),                          INTENT(IN   ) :: ErrStatLocal                  ! Error status of the operation
+    CHARACTER(KIND=C_CHAR, LEN=ErrMsgLen_C), INTENT(IN   ) :: ErrMessLocal                  ! Error message if ErrStat /= ErrID_None
+    INTEGER(C_INT),                          INTENT(INOUT) :: ErrStatGlobal                 ! Error status of the operation
+    CHARACTER(KIND=C_CHAR),                  INTENT(INOUT) :: ErrMessGlobal(ErrMsgLen_C)    ! Error message if ErrStat /= ErrID_None
+    CHARACTER(*),                            INTENT(IN   ) :: RoutineName                   ! Name of the routine error occurred in
+
+    IF ( ErrStatLocal == ErrID_None ) RETURN
+
+    IF (ErrStatGlobal /= ErrID_None) THEN
+        ! print *, "in if", ErrStatGlobal, ErrID_None
+        ! ErrMessGlobal = TRIM(ErrMessGlobal)//new_line('a')
+        ! print *, "ErrMessGlobal", ErrMessGlobal
+    ENDIF
+    ! TODO: Does the line below need to remove the null char from ErrMessGlobal prior to appending to it?
+    ErrMessGlobal = TRANSFER( ErrMessGlobal//TRIM(RoutineName)//':'//TRIM(ErrMessLocal)//C_NULL_CHAR, ErrMessGlobal )
+    ! ErrMessGlobal = TRIM(ErrMessGlobal)//TRIM(RoutineName)//':'//TRIM(ErrMessLocal)
+    ErrStatGlobal = MAX(ErrStatGlobal, ErrStatLocal)
+
+END SUBROUTINE
+
+SUBROUTINE StringConvert_F2C(F_String, C_String, Length)
```

Comment:
Is GPL v3.0 compatible for redistribution under Apache 2.0?  We might not be able to take this routine as is.

Reply:
For reasons related to the design of the module, I ended up modifying this subroutine quite a bit. It's no longer copying it from the source, but it is inspired by that repository and I've noted this in the docstring.

Reply:
So I think there's no longer a license compatibility issue.